src/templates/next/components/Navbar/Navbar.tsx

```diff
@@ -1,178 +1,285 @@
-                  <div className="flex flex-shrink-0 items-center">
-                    <LinkComponent href="/">
-                      <img
-                        className="h-8 w-auto"
-                        src={logo.url}
-                        alt={logo.alt}
-                      />
+    <div className="flex flex-col w-full">
+      <div className="flex flex-row gap-5 px-14 xl:px-0 py-6 mx-auto max-h-24 w-full xl:max-w-screen-xl">
+        {/* Logo */}
+        <LinkComponent href="/">
+          <img
+            src={logoUrl}
+            alt={logoAlt}
+            className="h-full w-full max-w-48 object-cover object-center"
+          />
+        </LinkComponent>
+
+        {/* Spacer */}
+        <div className="flex-1" />
+
+        {/* Search */}
+        {search && search.type === "localSearch" && !isHamburgerOpen && (
+          <form
+            action={search.searchUrl}
+            method="get"
+            className={`flex flex-row gap-2 h-10 ${
+              isSearchOpen ? "hidden lg:flex" : ""
+            }`}
+          >
+            {isSearchOpen && (
+              <input
+                type="search"
+                name="q"
+                placeholder="Search this site"
+                className="w-96 px-4 py-2 border border-divider-medium focus:border-site-primary focus:ring-site-primary focus:outline-none hidden lg:block"
+              />
+            )}
+            <button
+              onClick={(e) => {
+                if (!isSearchOpen) {
+                  e.preventDefault()
+                  setIsSearchOpen(true)
+                }
+              }}
+              type="submit"
+              aria-label="Search this site"
+            >
+              <BiSearch className="text-2xl mt-0.5" />
+            </button>
+          </form>
+        )}
+
+        {/* Hamburger menu for small screens */}
+        <button className="block xl:hidden h-10">
+          {isHamburgerOpen ? (
+            <button
+              onClick={() => {
+                setIsHamburgerOpen(!isHamburgerOpen)
+                setNavItemOpen(-1)
+              }}
+              aria-label="Close navigation menu"
+            >
+              Close
+              <BiX className="inline ml-1 -mt-0.5 text-2xl" />
+            </button>
+          ) : (
+            <button
+              onClick={() => setIsHamburgerOpen(!isHamburgerOpen)}
+              aria-label="Open navigation menu"
+            >
+              <BiMenu className="text-2xl mt-1.5" />
+            </button>
+          )}
+        </button>
+      </div>
+
+      {/* Search bar (for mobile/tablet) */}
+      <div className="block lg:hidden">
+        {isSearchOpen && search && search.type === "localSearch" && (
+          <form action={search.searchUrl} method="get">
+            <div className="flex flex-row gap-4 h-10 px-14">
+              <input
+                type="search"
```

Comment:
technically `input`s are supposed to have a corresponding `label`. (see [here](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label)) 

i only mention this because it seems liek we care abuot a11y

Reply:
Added `title` and `aria-label` instead, recommended in https://stackoverflow.com/a/23151986